packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+    const deploymentType = lustreConfiguration.deploymentType;
+
+    this.validateExportPath(lustreConfiguration.exportPath, lustreConfiguration.importPath);
+    this.validateImportedFileChunkSize(lustreConfiguration.importedFileChunkSize);
+    this.validateImportPath(lustreConfiguration.importPath);
+    this.validatePerUnitStorageThroughput(deploymentType, lustreConfiguration.perUnitStorageThroughput);
+    this.validateStorageCapacity(deploymentType, props.storageCapacity);
+    this.validateWeeklyMaintenanceStartTime(lustreConfiguration.weeklyMaintenanceStartTime);
+  }
+
+  /**
+   * Validates the export path is in the correct format and matches the import path.
+   */
+  private validateExportPath(exportPath?: string, importPath?: string): void {
+    if (exportPath === undefined) { return; }
+
+    if (importPath === undefined) {
+      throw new Error('Cannot define an export path without also defining an import path');
+    }
+    if (!exportPath.startsWith(importPath)) {
+      throw new Error(`The export path "${exportPath}" is invalid. Expecting the format: s3://{IMPORT_PATH}/optional-prefix`);
+    }
+    if (exportPath.length > 900) {
+      throw new Error(`The export path "${exportPath}" exceeds the maximum length of 900 characters`);
+    }
+  }
+
+  /**
+   * Validates the importedFileChunkSize is in the correct range.
+   */
+  private validateImportedFileChunkSize(importedFileChunkSize?: number): void {
+    if (importedFileChunkSize === undefined) { return; }
+
+    if (importedFileChunkSize < 1 || importedFileChunkSize > 512000) {
+      throw new Error(`importedFileChunkSize cannot be ${importedFileChunkSize} MiB. It must be a value from 1 to 512,000 MiB`);
+    }
+  }
+
+  /**
+   * Validates the import path is the correct format.
+   */
+  private validateImportPath(importPath?: string): void {
+    if (importPath === undefined) { return; }
+
+    const regexp = /^s3:\/\//;
+
+    if (importPath.search(regexp) === -1) {
+      throw new Error(`The import path "${importPath}" is invalid. Expecting the format: s3://{BUCKET_NAME}/optional-prefix`);
+    }
+    if (importPath.length > 900) {
+      throw new Error(`The import path "${importPath}" exceeds the maximum length of 900 characters`);
+    }
+  }
+
+  /**
+   * Validates the perUnitStorageThroughput is defined correctly for the given deploymentType.
+   */
+  private validatePerUnitStorageThroughput(deploymentType: FsxLustreDeploymentType, perUnitStorageThroughput?: number) {
+    if (perUnitStorageThroughput === undefined) { return; }
+
+    if (deploymentType === FsxLustreDeploymentType.PERSISTENT_1) {
+      switch (perUnitStorageThroughput) {
+        case 50:
+        case 100:
+        case 200:
+          break;
+        default:
+          throw new Error('perUnitStorageThroughput must be 50, 100, or 200 MB/s/TiB');
+      }
+    } else {
+      throw new Error('perUnitStorageThroughput can only be set for the PERSISTENT_1 deployment type');
```

Comment:
Please change to early return style. Save the indentation!

Reply:
Fixed